main.go

```diff
@@ -86,68 +86,69 @@ func setupManager(log logr.Logger, opts options.Options) (manager.Manager, error
 		return nil, fmt.Errorf("getting config: %w", err)
 	}
 
-	log.Info("Initialize Scheme")
+	log.Info("Initialize Manager")
 
-	scheme, err := initializeScheme()
+	mgr, err := initManager(log, opts, cfg)
 	if err != nil {
-		return nil, fmt.Errorf("initializing scheme: %w", err)
+		return nil, fmt.Errorf("initializing manager: %w", err)
 	}
 
-	log.Info("Initialize Manager")
+	log.Info("Registering Components")
 
-	mgrOpts, err := getManagerOptions(scheme, opts)
-	if err != nil {
-		return nil, fmt.Errorf("getting manager options: %w", err)
-	}
+	log.Info("-> Initialize Prometheus Registry")
 
-	mgr, err := manager.New(cfg, mgrOpts)
+	reg, err := dvoProm.SetupRegistry()
 	if err != nil {
-		return nil, fmt.Errorf("initializing manager: %w", err)
+		return nil, fmt.Errorf("initializing Prometheus server: %w", err)
 	}
 
-	if err := mgr.AddHealthzCheck("health", healthz.Ping); err != nil {
-		return nil, fmt.Errorf("adding healthz check: %w", err)
+	log.Info(fmt.Sprintf("-> Initialize Prometheus metrics endpoint on %q", opts.MetricsEndpoint()))
```

Comment:
I think `->` is not necessary. I think the better way to distinguish it would be to use a log with different name or perhaps different log level. I don't know `zap` much....

Reply:
I was using that symbol as a bullet for the log, to have a minimal appraise of what was going on.
e.g.
`2023-08-22T12:31:11+02:00	INFO	DeploymentValidation	Load KubeConfig
2023-08-22T12:31:11+02:00	INFO	DeploymentValidation	Initialize Manager
2023-08-22T12:31:11+02:00	INFO	DeploymentValidation	-> Initialize Scheme
2023-08-22T12:31:11+02:00	INFO	DeploymentValidation	-> Getting Manager Options
2023-08-22T12:31:11+02:00	INFO	DeploymentValidation	-> Adding Healthz and Readyz checks
2023-08-22T12:31:11+02:00	INFO	DeploymentValidation	Registering Components
2023-08-22T12:31:11+02:00	INFO	DeploymentValidation	-> Initialize Prometheus Registry
2023-08-22T12:31:11+02:00	INFO	DeploymentValidation	-> Initialize Prometheus metrics endpoint on "http://0.0.0.0:8383/metrics"
2023-08-22T12:31:11+02:00	INFO	DeploymentValidation	-> Initialize ConfigMap watcher
2023-08-22T12:31:11+02:00	INFO	DeploymentValidation	-> Initialize Validation Engine
2023-08-22T12:31:11+02:00	INFO	DeploymentValidation	-> Initialize Reconciler
2023-08-22T12:31:11+02:00	INFO	DeploymentValidation	Starting Manager`

Reply:
Yes I know but I think this is lost when you have the log formatted in JSON.

Reply:
I was checking Zap library but nothing like this except for changing the name, and then you would have to change it back. It looks like a lot of work only for a little change.

Rolling back. :+1: